### PR TITLE
doc/v4 codemod

### PR DESCRIPTION
**Pull request by vagusX**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

add doc for v4 codemod

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | add doc for v4 codemod |
| 🇨🇳 Chinese | 编写 v4 codemod 的文档 |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered docs/react/codemod.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/codemod.en-US.md)
[View rendered docs/react/codemod.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/codemod.zh-CN.md)
[View rendered docs/react/contributing.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/contributing.en-US.md)
[View rendered docs/react/contributing.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/contributing.zh-CN.md)
[View rendered docs/react/customize-theme.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/customize-theme.en-US.md)
[View rendered docs/react/customize-theme.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/customize-theme.zh-CN.md)
[View rendered docs/react/faq.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/faq.en-US.md)
[View rendered docs/react/faq.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/faq.zh-CN.md)
[View rendered docs/react/getting-started.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/getting-started.en-US.md)
[View rendered docs/react/getting-started.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/getting-started.zh-CN.md)
[View rendered docs/react/i18n.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/i18n.en-US.md)
[View rendered docs/react/i18n.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/i18n.zh-CN.md)
[View rendered docs/react/practical-projects.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/practical-projects.en-US.md)
[View rendered docs/react/practical-projects.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/practical-projects.zh-CN.md)
[View rendered docs/react/recommendation.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/recommendation.en-US.md)
[View rendered docs/react/recommendation.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/recommendation.zh-CN.md)
[View rendered docs/react/use-in-typescript.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/use-in-typescript.en-US.md)
[View rendered docs/react/use-in-typescript.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/use-in-typescript.zh-CN.md)
[View rendered docs/react/use-with-create-react-app.en-US.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/use-with-create-react-app.en-US.md)
[View rendered docs/react/use-with-create-react-app.zh-CN.md](https://github.com/ant-design/ant-design/blob/doc/v4-codemod/docs/react/use-with-create-react-app.zh-CN.md)

### CODE REVIEW

File: docs/react/codemod.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+order: 1
+title: V3 to V4
```
Comment (zombieJ) [inline docs/react/codemod.zh-CN.md]: 从 v3 到 v4

File: docs/react/codemod.en-US.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
+---
+order: 1
+title: V3 to V4
+---
+
+Ant design v4 is coming, we provide a codemod cli, [@ant-design/codemod-v4](https://github.com/ant-design/codemod-v4), a collection of codemod scripts that help upgrade antd v4 using [jscodeshift](https://github.com/facebook/jscodeshift).
```
Comment (zombieJ) [inline docs/react/codemod.en-US.md]: `is coming` 不用，否则正式版我们还得再改一遍

File: docs/react/codemod.en-US.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+```shell
+# global installation
+npm i -g @ant-design/codemod-v4
+antd4-codemod src
+
+# use npx
+npx -p @ant-design/codemod-v4 antd4-codemod src
+```
+
+## Codemod scripts introduction
```
Comment (zombieJ) [inline docs/react/codemod.en-US.md]: 添加说明，codemod 可以自动将 v3 中被废弃的 api 迁移到兼容包版本 balabala。以下是示例，注意，大部分情况下你不需要手工进行迁移！

File: docs/react/codemod.en-US.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+Before run codemod scripts, you'd better make sure to commit your local git changes firstly.
+
+```shell
+# global installation
+npm i -g @ant-design/codemod-v4
+antd4-codemod src
+
+# use npx
+npx -p @ant-design/codemod-v4 antd4-codemod src
```
Comment (zombieJ) [inline docs/react/codemod.en-US.md]: 执行 `npx`  不需要全局安装 `@ant-design/codemod-v4`

File: docs/react/codemod.en-US.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+## Codemod scripts introduction
+
+#### `v3-Component-to-compatible`
```
Comment (zombieJ) [inline docs/react/codemod.en-US.md]: 换成更口语化的描述如：Migrate removed component into compatible package

Comment (zombieJ) [reply docs/react/codemod.en-US.md]: 下同

File: docs/react/codemod.en-US.md (hunk 2 above)
Comment (vagusX) [reply docs/react/codemod.en-US.md]: launched?

File: docs/react/codemod.en-US.md (hunk 4 above)
Comment (vagusX) [reply docs/react/codemod.en-US.md]: 加一个 or use npx

File: docs/react/codemod.en-US.md (hunk 5 above)
Comment (vagusX) [reply docs/react/codemod.en-US.md]: 啊？这个是 jscodeshift 脚本名？要改？感觉有点奇怪

File: docs/react/codemod.zh-CN.md (hunk 1 above)
Comment (vagusX) [reply docs/react/codemod.zh-CN.md]: ok

File: docs/react/codemod.en-US.md (hunk 3 above)
Comment (vagusX) [reply docs/react/codemod.en-US.md]: ok

File: docs/react/codemod.en-US.md (hunk 2 above)
Comment (zombieJ) [reply docs/react/codemod.en-US.md]: 不用提，直接说我们提供了 codemod 功能帮助你进行 v3 到 v4 的迁移。

File: docs/react/codemod.en-US.md (hunk 5 above)
Comment (zombieJ) [reply docs/react/codemod.en-US.md]: 这个脚本用户不直接接触的话，他们是不需要了解的

File: docs/react/codemod.en-US.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
+---
+order: 1
+title: V3 to V4
+---
+
+Ant design v4 is coming, we provide a codemod cli, [@ant-design/codemod-v4](https://github.com/ant-design/codemod-v4), a collection of codemod scripts that help upgrade antd v4 using [jscodeshift](https://github.com/facebook/jscodeshift).
+
+## Usage
+
+Before run codemod scripts, you'd better make sure to commit your local git changes firstly.
```
Comment (shaodahong) [inline docs/react/codemod.en-US.md]: 这个最好加粗强调下，不然没太注意到执行，本地未提交的很难恢复

File: docs/react/codemod.en-US.md (hunk 4 above)
Comment (shaodahong) [reply docs/react/codemod.en-US.md]: 感觉默认就应该 `npx` 比较好，也是现在主流的方式，然后可以 or global

Comment (afc163) [reply docs/react/codemod.en-US.md]: yarn 的用法也加上

File: docs/react/codemod.en-US.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
+---
+order: 1
+title: V3 to V4
```
Comment (afc163) [inline docs/react/codemod.en-US.md]: 这篇文章如果是这个标题，不能只有 codemod 的部分，还要有 antd 的 break change 点，需要手动升级的部分的说明，react 15 到 16 的升级点等等完整的说明。

Comment (afc163) [reply docs/react/codemod.en-US.md]: 文件名也要改成 migration-v4

Comment (zombieJ) [reply docs/react/codemod.en-US.md]: 名字改完后，我再加上 breaking change list。

File: docs/react/codemod.zh-CN.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+---
+
+Ant design v4 即将发布，我们提供了 codemod cicli 工具，[@ant-design/codemod-v4](https://github.com/ant-design/codemod-v4), 能帮助你快速升级到 v4 版本，基于 [jscodeshift](https://github.com/facebook/jscodeshift) 构建。
```
Comment (shaodahong) [inline docs/react/codemod.zh-CN.md]:
```suggestion
Ant design v4 即将发布，我们提供了 codemod cli 工具，[@ant-design/codemod-v4](https://github.com/ant-design/codemod-v4)，能帮助你快速升级到 v4 版本，基于 [jscodeshift](https://github.com/facebook/jscodeshift) 构建。
```

**Comment by shaodahong** [comment]
image
这个缩进

File: docs/react/codemod.en-US.md (hunk 6 above)
Comment (vagusX) [reply docs/react/codemod.en-US.md]: 并不会，本地有没有提交的代码的话，这个 cli 是不会执行的，内置了检测 git status，需要用户 commit 才会跑起来

**Comment by vagusX** [comment]
> image
> 这个缩进

这种标准缩进应该是啥，两个空格？ @shaodahong

**Comment by shaodahong** [comment]
> > image
> > 这个缩进
> 
> 这种标准缩进应该是啥，两个空格？

和下面的对齐？

File: docs/react/codemod.en-US.md (hunk 7 above)
Comment (vagusX) [reply docs/react/codemod.en-US.md]: 👌

File: docs/react/migration-v4.en-US.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+# use npm
+npm i -g @ant-design/codemod-v4
+# or use yarn
+yarn add -g @ant-design/codemod-v4
+
+# run
+antd4-codemod src
+
+# or use npx
+npx -p @ant-design/codemod-v4 antd4-codemod src
```
Comment (zombieJ) [inline docs/react/migration-v4.en-US.md]: npx 方案放最上面。global install 的放下面。

Comment (vagusX) [reply docs/react/migration-v4.en-US.md]: ok

**Comment by zombieJ** [comment]
CI failed. Merge `4.0-prepare`

**Comment by vagusX** [comment]
> CI failed. Merge `4.0-prepare`

done

File: docs/react/migration-v4.en-US.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+# or use yarn
+yarn add -g @ant-design/codemod-v4
+
+# run
+antd4-codemod src
+```
+
+## TL;DR
+
+`@ant-design/codemod-v4` will help you migrate to antd v4, `@ant-design/compatible` is the workaround for the deprecated APIs and components , so you don't need do it yourself in general. The blew content is the migration detail.
```
Comment (zombieJ) [inline docs/react/migration-v4.en-US.md]: `components ,` 多了一个空格

Comment (zombieJ) [reply docs/react/migration-v4.en-US.md]: > so you don't need do it yourself in general

In most case, you don't need to pay additional effort for migration manually.

> The blew content is the migration detail.

Follow is ...

File: docs/react/migration-v4.en-US.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+        onChange={onChange}
+        defaultValue={toContentState('@afc163')}
+        defaultSuggestions={['afc163', 'benjycui']}
+        onSelect={onSelect}
+      />
+    </div>
+  );
+```
+
+#### Update component which contains string icon props with specific v4 Icon component from `@ant-design/icons`
```
Comment (zombieJ) [inline docs/react/migration-v4.en-US.md]: Replace components string `icon` prop with new `@ant-design/icons`

File: docs/react/migration-v4.en-US.md (hunk 10 above)
Comment (afc163) [reply docs/react/migration-v4.en-US.md]:
```diff
- so you don't need do it yourself in general
+ in general, you don't have to use it by yourself.
```

```diff
- blew
+ below
```

**Pull request merged**